aip-console-tools-cli/src/main/java/com/castsoftware/aip/console/tools/commands/OnboardApplicationCommand.java

```diff
@@ -0,0 +1,121 @@
+)
+@Slf4j
+@Getter
+@Setter
+public class OnboardApplicationCommand extends BasicCollable {
+    @CommandLine.Option(names = {"-n", "--app-name"},
+            paramLabel = "APPLICATION_NAME",
+            description = "The Name of the application to analyze",
+            required = true)
+    private String applicationName;
+
+    @CommandLine.Option(names = {"-f", "--file"},
+            paramLabel = "FILE",
+            description = "A local zip or tar.gz file OR a path to a folder on the node where the source if saved")
+    private File filePath;
+
+    @CommandLine.Option(names = "--node-name", paramLabel = "NODE_NAME",
+            description = "The name of the node on which the application will be created.")
+    private String nodeName;
+
+    @CommandLine.Option(names = "--domain-name", paramLabel = "DOMAIN_NAME",
+            description = "A domain is a group of applications. You may use domain to sort/filter applications. Will be created if it doesn't exists. No domain will be assigned if left empty.")
+    private String domainName;
+
+    @CommandLine.Option(names = {"-exclude", "--exclude-patterns"},
+            description = "File patterns(glob pattern) to exclude in the delivery, separated with comma")
+    private String exclusionPatterns;
+    @CommandLine.Option(names = {"--exclusion-rules"}, split = ",", type = ExclusionRuleType.class
+            , description = "Project's exclusion rules, separated with comma. Valid values: ${COMPLETION-CANDIDATES}")
+    private ExclusionRuleType[] exclusionRules;
+
+    @CommandLine.Option(names = {"-S", "--snapshot-name"},
+            paramLabel = "SNAPSHOT_NAME",
+            description = "The name of the snapshot to create")
+    private String snapshotName;
+
+    @CommandLine.Option(names = "--module-option"
+            , description = "Generates a user defined module option for either technology module or analysis unit module. Possible value is one of: full_content, one_per_au, one_per_techno (default: ${DEFAULT-VALUE})")
+    private ModuleGenerationType moduleGenerationType = ModuleGenerationType.FULL_CONTENT;
+
+    @CommandLine.Mixin
+    private SharedOptions sharedOptions;
+
+    private static final VersionInformation MIN_VERSION = VersionInformation.fromVersionString("2.8.0");
+
+    public OnboardApplicationCommand(RestApiService restApiService, JobsService jobsService, UploadService uploadService, ApplicationService applicationService) {
+        super(restApiService, jobsService, uploadService, applicationService);
+    }
+
+    @Override
+    protected Integer processCallCommand() throws Exception {
+
+        if (filePath == null) {
+            log.error("A valid file path required to perform the FAST SCAN operation");
+            return Constants.RETURN_MISSING_FILE;
+        }
+        CliLogPollingProviderImpl logPollingProvider = new CliLogPollingProviderImpl(jobsService,
```

Comment:
Before this, you should check if onboarding is enabled on AIP Console, and send an error if that is not the case (like I said in another comment).
It is wasteful to create a Polling Provider and the Properties and call the ApplicationService, if you are just going to discard all this because Onboarding is not enabled in Console

Reply:
Ok